### PR TITLE
Prevent remove storage flag when there are something stored

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/1226

### CODE REVIEW

File: neo/SmartContract/InteropService.NEO.cs [hunk 1]
```diff
@@ -225,6 +225,7 @@ private static bool Contract_Update(ApplicationEngine engine)
                 contract = engine.Snapshot.Contracts.GetAndChange(contract.ScriptHash);
                 contract.Manifest = ContractManifest.Parse(manifest);
                 if (!contract.Manifest.IsValid(contract.ScriptHash)) return false;
+                if (!contract.HasStorage && engine.Snapshot.Storages.Find(engine.CurrentScriptHash.ToArray()).Any()) return false;
```
Comment (lock9) [inline neo/SmartContract/InteropService.NEO.cs]: Hi @shargon, this looks ok, but is there any need to disable storage? If the user already paid for this feature, why would him disable it?

Comment (shargon) [reply neo/SmartContract/InteropService.NEO.cs]: Scams or bad intentions...

Comment (shargon) [reply neo/SmartContract/InteropService.NEO.cs]: You can create again the contract, and start with information

Comment (lock9) [reply neo/SmartContract/InteropService.NEO.cs]: Can you explain to me a case where disabling the storage can benefit anyone? I believe it is a “ok” change, from a business logic perspective, it seems to make sense.

Comment (shargon) [reply neo/SmartContract/InteropService.NEO.cs]: - ScammerICO create a tx when he create a dummy smart contract.
- ScammerICO set the storages as he want (ownerAddress.Balance=999999999).
- He migrate the dummy SC, to the ICO smart contract
- Change the storage flag in the ICO smart contract
- Destroy it (remaining the information).
- Two months later.
- ScammerICO, start the ICO, creating the SmartContract
- Automatically, it takes the previous stored information, because was not removed.
- Investors, and explorer has no idea about what happen, but a new smart contract, without TX, have iregular account, hard to follow, and hard to trace.

Comment (shargon) [reply neo/SmartContract/InteropService.NEO.cs]: @lock9 Not similar, but also ... it could produce scenarios like this https://medium.com/quantstamp/exploiting-the-interface-of-etherscan-for-ethereum-attacks-17b72d2897e0
